lib/vector-core/buffers/src/disk/leveldb_buffer/reader.rs

```diff
@@ -108,14 +114,51 @@ where
             // This will usually complete instantly, but in the case of a large
             // queue (or a fresh launch of the app), this will have to go to
             // disk.
-            tokio::task::block_in_place(|| {
-                this.buffer.extend(
-                    this.db
-                        .iter(ReadOptions::new())
-                        .from(&Key(this.read_offset))
-                        .take(100),
-                );
-            });
+            loop {
+                match this.pending_read.take() {
+                    None => {
+                        // We have no pending read in-flight, so queue one up.
+                        let db = Arc::clone(&this.db);
+                        let read_offset = this.read_offset;
+                        let handle = tokio::task::spawn_blocking(move || {
+                            db.iter(ReadOptions::new())
+                                .from(&Key(read_offset))
+                                .take(1000)
```

Comment:
Not a big deal, but just noticing that this changed from 100 to 1000. Is that intentional?

Reply:
Sans comment I couldn't tell you. I would assume this is either an artifact of testing but whether it's a tag-along or an intentional change I don't know.